src/main/webapp/app/exam/participate/exercises/exam-submission.component.ts

```diff
@@ -1,14 +1,29 @@
 import { Submission } from 'app/entities/submission.model';
 import { Exercise } from 'app/entities/exercise.model';
+import { ChangeDetectorRef } from '@angular/core';
 
 export abstract class ExamSubmissionComponent {
     abstract hasUnsavedChanges(): boolean;
     abstract updateSubmissionFromView(): void;
+    abstract updateViewFromSubmission(): void;
```

Comment:
Why is this needed as an abstract method? I see it just used inside the inherited components for modeling, quiz and text exercises and not at all for programming exercises.

Reply:
I think @anditurdiu added it, because it makes sense from the interface design perspective, because every ExamSubmissionComponent needs to be able to update it's content from the given submission. The code-editor is an exception, because it handles updating of the view from the submission on it's own. I see your point, that it is not needed in this case. I think it is a question on how the responsibilities of the components are in the concept. As I understood, we have the exam-participation-component, which manages all the participations and the submission components for each exercise type are just handling the submissions. Therefore the programming editor is the exception here, by handling the participation and the submission, but it is okay to add it to the interface, because future submission components very likely need to implement this method as well.

Reply:
Ok